syntax = "proto2";


// serial name 'com.jpd.methodcards.data.library.MethodsProto'
message MethodsProto {
  // WARNING: a default value decoded when value is missing
  repeated MethodProto methods = 1;
}

// serial name 'com.jpd.methodcards.data.library.MethodProto'
message MethodProto {
  required string name = 1;
  required string notation = 2;
  required int32 stage = 3;
  // WARNING: a default value decoded when value is missing
  optional int32 ruleoffsFrom = 4;
  required int32 ruleoffsEvery = 5;
  // WARNING: a default value decoded when value is missing
  optional StandardCalls standardCalls = 6;
  // WARNING: a default value decoded when value is missing
  repeated CallProto customCalls = 9;
  required int32 magic = 7;
  required MethodClassificationProto classification = 8;
  // WARNING: a default value decoded when value is missing
  optional bool nameHasClassification = 10;
}

// serial name 'com.jpd.methodcards.data.library.MethodProto.CallProto'
message CallProto {
  required string name = 6;
  required string symbol = 1;
  required string notation = 2;
  // WARNING: a default value decoded when value is missing
  optional int32 from_ = 3;
  required int32 every = 4;
  required int32 cover = 5;
}

// serial name 'com.jpd.methodcards.data.library.MethodProto.StandardCalls'
message StandardCalls {
  required string bobNotation = 1;
  required string singleNotation = 2;
  required int32 every = 3;
}

// serial name 'com.jpd.methodcards.data.library.MethodProto.MethodClassificationProto'
enum MethodClassificationProto {
  None = 0;
  TreblePlace = 1;
  Delight = 2;
  Bob = 3;
  Jump = 4;
  Alliance = 5;
  Hybrid = 6;
  TrebleBob = 7;
  Place = 8;
  Surprise = 9;
}
